integration_tests/e2e/refer/programmeHistory.cy.ts

```diff
@@ -283,121 +283,148 @@ context('Programme history', () => {
     })
 
     describe('and adding details', () => {
-      const courseParticipation = courseParticipationFactory
-        .withCourseId()
-        .build({ courseId: courses[0].id, prisonNumber: person.prisonNumber })
-      const path = referPaths.programmeHistory.details.show({
-        courseParticipationId: courseParticipation.id,
-        referralId: referral.id,
-      })
+      describe('for a new entry', () => {
```

Comment:
I think the explicit test for an existing entry is needed, but don't all of the test expectations in the new entry apply for existing entries too? E.g. `updates the entry...` and `displays an error...`: I wonder if it's worth denesting them to make it clear they're the expectation whether or not the participation already has details? The participation stub could just be overwritten in the existing data test

Reply:
Yeah good shout.